src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java

```diff
@@ -349,17 +350,69 @@ private static boolean isSingleMap(Map<String, ?> args) {
         return Jenkins.getActiveInstance().getDescriptorByName(id);
     }
 
-    @Restricted(DoNotUse.class)
-    public Collection<? extends StepDescriptor> getStepDescriptors(boolean advanced) {
-        TreeSet<StepDescriptor> t = new TreeSet<StepDescriptor>(new StepDescriptorComparator());
+    @Restricted(NoExternalUse.class)
+    public Collection<QuasiDescriptor> getQuasiDescriptors(boolean advanced) {
+        TreeSet<QuasiDescriptor> t = new TreeSet<>();
         for (StepDescriptor d : StepDescriptor.all()) {
             if (d.isAdvanced() == advanced) {
-                t.add(d);
+                t.add(new QuasiDescriptor(d));
+                if (d.isMetaStep()) {
+                    DescribableModel<?> m = new DescribableModel<>(d.clazz);
+                    Collection<DescribableParameter> parameters = m.getParameters();
+                    if (parameters.size() == 1) {
+                        DescribableParameter delegate = parameters.iterator().next();
+                        if (delegate.isRequired()) {
```

Comment:
No `getSoleRequiredParameter` is inappropriate here, because it would return non-null for `checkout` if made a metastep, which is not a case we are able to handle yet. See the follow-up issue.